### PR TITLE
feat: add auto archiving action

### PR DESCRIPTION
**Pull request by PierreDelpy**
The goal is to automatically archive unused branches (rename to "archived/..." after 365 days of no commit).
Actual code execution is [commented](https://github.com/samply/bridgehead/pull/238/files#diff-d9915e153125f24788b5dfc941e10c009a508dee3bf93ecf09ded01ab27189afR31-R32) (since I don't want to trigger the process until approved) 

These branches would be affected:
https://github.com/samply/bridgehead/actions/runs/11340581646/job/31537252148#step:5:10

Will delete this code once ready for merging

### CODE REVIEW

**Review by lablans — COMMENTED** [review]
Isn't this something we should do manually? Maybe instead of renaming the branch, rather send an e-mail / alert?

Also, some Python comments in the code.

**Comment by PierreDelpy** [comment]
> Isn't this something we should do manually? Maybe instead of renaming the branch, rather send an e-mail / alert?

Yes we should definitely do this manually. Therefore, I added the 365 days of inactivity. However, some branches are forgotten and I think with our currently 32 branches, we need an additional mechanism.

File: .github/scripts/rename_inactive_branches.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Check if the branch is inactive
+def is_inactive(commit_url):
+    last_commit_date = requests.get(commit_url, headers=HEADERS).json()['commit']['committer']['date']
+    return datetime.strptime(last_commit_date, '%Y-%m-%dT%H:%M:%SZ') < CUTOFF_DATE
+
+# Rename inactive branches
+for branch in get_branches():
+    if is_inactive(branch['commit']['url']):
+        #rename_branch(branch['name'], f"archived/{branch['name']}")
+        print(f"[LOG] Branch '{branch['name']}' is inactive and would be renamed to 'archived/{branch['name']}'")
```
[the remark this answers is not in the document]
Comment (PierreDelpy) [reply .github/scripts/rename_inactive_branches.py]: > Add `if __name__ == "__main__":`

Fixed

File: .github/scripts/rename_inactive_branches.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+REPO = 'samply/bridgehead'
+HEADERS = {'Authorization': f'token {GITHUB_TOKEN}', 'Accept': 'application/vnd.github.v3+json'}
+API_URL = f'https://api.github.com/repos/{REPO}/branches'
+INACTIVE_DAYS = 365
+CUTOFF_DATE = datetime.now() - timedelta(days=INACTIVE_DAYS)
+
+# Fetch all branches
+def get_branches():
+    response = requests.get(API_URL, headers=HEADERS)
+    return response.json() if response.status_code == 200 else []
```
[the remark this answers is not in the document]
Comment (PierreDelpy) [reply .github/scripts/rename_inactive_branches.py:17]: Fixed in https://github.com/samply/bridgehead/pull/238/commits/a04d09e727e15f3bfb3c49b210bd21f523b2bd92
Is this fine for you?

File: .github/scripts/rename_inactive_branches.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+# Rename inactive branches
+def rename_branch(old_name, new_name):
+    rename_url = f'https://api.github.com/repos/{REPO}/branches/{old_name}/rename'
+    response = requests.post(rename_url, json={'new_name': new_name}, headers=HEADERS)
+    print(f"Renamed branch {old_name} to {new_name}" if response.status_code == 201 else f"Failed to rename {old_name}: {response.status_code}")
```
[the remark this answers is not in the document]
Comment (PierreDelpy) [reply .github/scripts/rename_inactive_branches.py:24]: Fixed in https://github.com/samply/bridgehead/pull/238/commits/a04d09e727e15f3bfb3c49b210bd21f523b2bd92
Is this fine for you?

File: .github/scripts/rename_inactive_branches.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lablans) [reply .github/scripts/rename_inactive_branches.py:17]: yep

File: .github/scripts/rename_inactive_branches.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lablans) [reply .github/scripts/rename_inactive_branches.py:24]: yep

**Review by lablans — APPROVED** [review]

File: .github/workflows/rename-inactive-branches.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+name: Cleanup - Rename Inactive Branches
+
+on:
+  schedule:
+    - cron: '0 0 * * 0'  # Runs every Sunday at midnight
+  #todo delete; only trigger for testing
+  pull_request:
+    branches:
+      - develop
```
Comment (TKussel) [inline .github/workflows/rename-inactive-branches.yml:9]: This is not required anymore, no?

**Pull request merged**